src/SessionService.php

```diff
@@ -93,15 +94,46 @@ public function commitSession(SessionData $session, ResponseInterface $response)
         $data = $session->getData();
 
         if (count($data) === 0) {
-            $this->storage->destroy($session_id);
-        } else {
-            $this->storage->write($session_id, $data, $this->ttl);
+            if (! $session->isNew()) {
+                $this->storage->destroy($session_id);
+
+                return $response->withAddedHeader(self::SET_COOKIE_HEADER, $this->createCookie("", 0));
+            }
+
+            return $response;
+        }
+
+        $this->storage->write($session_id, $data, $this->ttl);
+
+        if ($session->isNew()) {
+            $response = $response->withAddedHeader(
+                self::SET_COOKIE_HEADER,
+                $this->createCookie($session_id, $this->getTime() + $this->ttl)
+            );
         }
 
-        $secure = $this->secure_only ? " Secure;" : "";
+        return $response;
+    }
+
+    protected function getTime(): int
+    {
+        return time();
+    }
 
-        $header = sprintf(self::COOKIE_NAME . "=%s; Path=/; HTTPOnly; SameSite=Lax;%s", $session_id, $secure);
+    private function createCookie(string $value, int $expires): string
+    {
+        $cookie = [
+            self::COOKIE_NAME . "={$value}",
+            "Path=/",
+            "HTTPOnly",
+            "SameSite=Lax",
+            "Expires=" . DateTime::createFromFormat("U", $expires, timezone_open('UTC'))->format(DateTime::COOKIE)
```

Comment:
Do we consider this, now we are setting the time on the cookie: 

> Types of Cookies
There are two different types of cookies - session cookies and persistent cookies. If a cookie does not contain an expiration date, it is considered a session cookie. Session cookies are stored in memory and never written to disk. When the browser closes, the cookie is permanently lost from this point on. If the cookie contains an expiration date, it is considered a persistent cookie. On the date specified in the expiration, the cookie will be removed from the disk.

https://www.cisco.com/c/en/us/support/docs/security/web-security-appliance/117925-technote-csc-00.html#toc-hId-841855014